src/datamancer/io.nim

```diff
@@ -607,6 +566,73 @@ proc parseCsvString*(csvData: string,
   result = readCsvTypedImpl(data, csvData.len, countNonEmptyLines(csvData),
                             sep, header, skipLines, toSkip, colNames)
 
+proc readCsvFromUrl(url: string,
+              sep: char = ',',
+              header: string = "",
+              skipLines = 0,
+              toSkip: set[char] = {},
+              colNames: seq[string] = @[]
+             ): DataFrame =
+  ## Reads a DF from a web URL (which must contain a CSV file)
+  var client = newHttpClient()
+  return parseCsvString(client.getContent(url), sep, header, skipLines, toSkip, colNames)
+
+proc readCsv*(fname: string,
+              sep: char = ',',
+              header: string = "",
+              skipLines = 0,
+              toSkip: set[char] = {},
+              colNames: seq[string] = @[],
+              skipInitialSpace = true,
+              quote = '"',
+             ): DataFrame =
+  ## Reads a DF from a CSV file or a web URL using the separator character `sep`.
+  ## 
+  ## `fname` can be a local filename or a web URL. If `fname` starts with
+  ## "http://" or "https://" the file contents will be read from the selected
+  ## web server. No catching is performed so if you plan to read from the same
+  ## URL multiple times it might be best to download the file manually instead.
+  ## Please note that to download files from https URLs you must compile with
+  ## the -d:ssl option. Also note that the `skipInitialSpace` and `quote`
+  ## arguments are ignored when reading from a web URL.
```

Comment:
Given that the missing `skipInitialSpace` and `quote` is just an oversight feel free to fix it (or I'll do it in a follow up for this and the regular case.